corehq/apps/reports/dispatcher.py

```diff
@@ -311,7 +312,11 @@ class AdminReportDispatcher(ReportDispatcher):
     map_name = 'ADMIN_REPORTS'
 
     def permissions_check(self, report, request, domain=None, is_navigation_check=False):
-        return hasattr(request, 'couch_user') and request.user.has_perm("is_superuser")
+        return (
+            hasattr(request, 'couch_user')
+            and request.user.has_perm("is_superuser")
+            and not is_request_using_sso(request)
+        )
```

Comment:
I don't think this is the right place to do this check -- reports shouldn't need to know about SSO restrictions. I think this would lead to a developer having to know that any place we wanted to check for superuser privileges, we would also have to remember to make sure they weren't using SSO. I think a cleaner solution would be to combine this functionality into a request method, so we only have to remember it in one place